src/model/trip-points-model.js

```diff
@@ -1,34 +1,79 @@
-import { mockPoints } from '../mock/mock-points';
+
 import Observable from '../framework/observable';
+import { UpdateType } from '../const';
 
 export default class TripPointsModel extends Observable {
-  constructor() {
+  #pointsApiService = null;
+  #points = new Map();
+
+  #isReady = false;
+
+  constructor({
+    pointsApiService
+  }) {
     super();
-    mockPoints.forEach((point)=>{
-      this.#points.set(
-        point.id, {
-          'base_price': point.base_price,
-          'date_from':  point.date_from,
-          'date_to': point.date_to,
-          'destination': point.destination,
-          'is_favorite': point.is_favorite,
-          'offers': point.offers,
-          'type': point.type
-        });
 
-    });
+
+    this.#pointsApiService = pointsApiService;
+
+
   }
 
-  #points = new Map();
+  async init () {
+    try {
+      const points = await this.#pointsApiService.points;
+      points.forEach((point)=>this.#points.set(this.#adaptPointToClient(point)[0],this.#adaptPointToClient(point)[1]));
```

Comment:
```suggestion
      points.forEach((point) => this.#points.set(point.id, this.#adaptPointToClient(point)[1]));
```
Ну или сохранить сначала поинт в клиентском формате в переменную, чтобы два раза не конвертировать.